spec/requests/missions_spec.rb

```diff
@@ -34,29 +25,29 @@
 
   describe 'to show an event' do
     it 'renders successfully the template' do
-      event = create_event
+      event = create :mission, event: true
 
       get mission_path(event.id)
 
       expect(response).to be_successful
     end
 
     it 'displays the name of event' do
-      event = create_event
+      event = create :mission, event: true
 
       get mission_path(event.id)
 
       expect(response.body).to include(event.name.capitalize)
     end
 
-    context 'when a member particpate in an event' do
+    context 'when a member participate in an event' do
       it "displays the full name's member" do
-        event = create_event
-        create :participation, event_id: event.id, participant_id: super_admin.id
+        event = create :mission, event: true
+        create :participation, event_id: event.id, participant_id: member.id
 
         get mission_path(event.id)
 
-        expect(response.body).to include("#{super_admin.first_name} #{super_admin.last_name}")
+        expect(response.body).to include("#{member.first_name} #{member.last_name}")
       end
     end
   end
```

Comment:
```ruby
describe 'GET show' do
  subject(:get_show) { get mission_path(mission) }

  context 'when the mission is an :event' do
    let(:mission) { create_event }
    it 'has a successful response' do
      get_show
      expect(response).to be_successful
    end
    it 'displays the name of the event' do
      ...
    end
    context 'when a member participates in an event' do
      it 'displays the blabla...' do
        create :participation, event_id: mission.id, participant_id: member.id

        get_show

        expect(.....
      end
    end
  end

  context 'when the mission is not an event' do
    it 'returns a successful response' do
      get_show
      expect(response).to be_successful
    end

    ...etc...

  end
end
```

The point here is to regroup REST requests by type, and then you can detail that request with a context. It also allows us to use the `subject` for our `describe`, that is common to all our tests cases. This improves readability, as we have less to read, but the test's story is still obvious with this small abstraction